mechanisms/generate_catalogue

```diff
@@ -77,7 +83,7 @@ def parse_arguments():
     return vars(parser.parse_args())
 
 
-def generate(modpfx='', arbpfx='', modules=[], backends=[], **rest):
+def generate(modpfx='', arbpfx='', modules=[], backends=[], catalogue='default', **rest):
```

Comment:
Should we make the catalogue name default `'default'`, or force the user to always specify a name.
My preference is to not have a default name.

Reply:
Agreed